### PR TITLE
add link for shell activation failure warning

### LINKED ISSUE #148727: add hover link to docs when shell integration is disabled due to complex `PROMPT_COMMAND`
When a `PROMPT_COMMAND` contains both a space and semi colon, we disable shell integration and show a warning. On hover, the warning should have a link to the docs that explain what can be done about this.

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #148727

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts [hunk 1]
```diff
@@ -26,18 +40,19 @@ export class TerminalShellIntegrationLinkDetector implements ITerminalLinkDetect
...
 			return [{
-				text: linkText,
+				text: match,
 				type: TerminalBuiltinLinkType.Url,
-				label: localize('learn', 'Learn about shell integration'),
+				label: match === LinkType.Activated ? localize('learn', 'Learn about shell integration') : localize('read more', 'Read more about this warning'),
 				uri: URI.from({
 					scheme: Schemas.https,
-					path: 'aka.ms/vscode-shell-integration'
+					path: match === LinkType.Activated ? 'aka.ms/vscode-shell-integration' : 'aka.ms/vscode-shell-integration#_complex-bash-promptcommand'
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts:51]: @Tyriar how do I make a custom redirect like this?

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts:51]: Sent you a dm

File: src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts [hunk 2]
```diff
@@ -50,18 +65,29 @@ export class TerminalShellIntegrationLinkDetector implements ITerminalLinkDetect
...
+		}
+		let j: number = 0;
+		for (; j < disabledLink.codes.length; j++) {
+			cell = lines[Math.floor(j / this.xterm.cols)].getCell(j % this.xterm.cols, cell);
+			if (cell?.getCode() !== disabledLink.codes[j]) {
+				break;
 			}
 		}
-		return true;
+		return j === disabledLink.codes.length ? LinkType.Disabled : undefined;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts:90]: You could add a `const links = [activatedLink, disabledLink]` at the top level and then use `for (const link of links)` here to reduce duplication

File: src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts [hunk 3]
```diff
@@ -50,18 +65,29 @@ export class TerminalShellIntegrationLinkDetector implements ITerminalLinkDetect
 		return [];
 	}
 
-	private _matches(lines: IBufferLine[]): boolean {
-		if (lines.length < linkCodes.length) {
-			return false;
+	private _matches(lines: IBufferLine[]): LinkType | undefined {
+		if (lines[0].length < activatedLink.codes.length && lines[0].length < disabledLink.codes.length) {
+			return undefined;
 		}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/links/terminalShellIntegrationLinkDetector.ts:71]: At top level `const minLinkLength = Math.min(...links.map(e => e.length))`? Then you could do this:

```ts
if (lines[0].length < minLinkLength) {
	return undefined;
}
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**